crates/cheatcodes/src/script.rs

```diff
@@ -72,6 +73,15 @@ pub struct Broadcast {
     pub single_call: bool,
 }
 
+/// Contains context for wallet management.
+#[derive(Debug)]
+pub struct ScriptWalletsData {
+    /// All signers in scope of the script.
+    pub multi_wallet: MultiWallet,
+    /// Optional signer provided as `--sender` flag.
+    pub provided_sender: Option<Address>,
+}
```

Comment:
I see we're using `Arc<Mutex<ScriptWalletsData>` a few times,

can we have a wrapper type for this

```rust
struct ScriptWallets {
  // maybe Option?
  inner: Arc<Mutex<ScriptWallets>>
}
```